### PR TITLE
GH-93899: fix checks for eventfd flags 

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-93899

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

**Review by erlend-aasland — CHANGES_REQUESTED** [review]
LGTM, but `os_eventfd_impl` could also use a wrapper, and some of the tests should skip if these flags are not present.

**Comment by kumaraditya303** [comment]
> but os_eventfd_impl could also use a wrapper,

Added guard to `os.eventfd`.

> and some of the tests should skip if these flags are not present.

The test are already skipped on older kernel versions.

https://github.com/python/cpython/blob/3c94d3395edbcd299cbdacd09660ed88e654eeef/Lib/test/test_os.py#L3686-L3690

I have made the requested changes; please review again

File: Modules/posixmodule.c [hunk 1]
```diff
@@ -15463,10 +15463,16 @@ all_ins(PyObject *m)
 #endif /* HAVE_MEMFD_CREATE */
 
 #ifdef HAVE_EVENTFD
+#ifdef EFD_CLOEXEC
```
Comment (tiran) [inline Modules/posixmodule.c]:
```suggestion
#if defined(HAVE_EVENTFD) && defined(EFD_CLOEXEC)
```

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -15463,10 +15463,16 @@ all_ins(PyObject *m)
...
     if (PyModule_AddIntMacro(m, EFD_CLOEXEC)) return -1;
+#endif
```
Comment (tiran) [inline Modules/posixmodule.c]:
```suggestion
```

File: Modules/posixmodule.c [hunk 3]
```diff
@@ -15463,10 +15463,16 @@ all_ins(PyObject *m)
...
+#ifdef EFD_NONBLOCK
     if (PyModule_AddIntMacro(m, EFD_NONBLOCK)) return -1;
+#endif
+#ifdef EFD_SEMAPHORE
     if (PyModule_AddIntMacro(m, EFD_SEMAPHORE)) return -1;
 #endif
+#endif
```
Comment (tiran) [inline Modules/posixmodule.c]:
```suggestion
#endif  /* HAVE_EVENTFD && EFD_CLOEXEC */
```

**Review by tiran — CHANGES_REQUESTED** [review]
If you actually want to support ancient Kernels, then let's use consistent checks.

**Review by erlend-aasland — APPROVED** [review]
LGTM. You may consider to make the NEWS entry a little bit more specific ("older kernel versions" is a bit vague).

**Comment by kumaraditya303** [comment]
> LGTM. You may consider to make the NEWS entry a little bit more specific ("older kernel versions" is a bit vague).

Thanks for the review! I have reworded news entry, should be better now!

**Comment by erlend-aasland** [comment]
Thanks, I'm fine with that. I'll land it as soon as the CI is green. Thanks!

**Review by tiran — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Will you fix the backports, Kumar? It's probably just a clinic regen that's missing. (Can we teach Miss I to do that?)

**Comment by kumaraditya303** [comment]
> Will you fix the backports, Kumar?

Done

**Pull request merged**